### PR TITLE
admin: add GitHub PR template.

### PR DESCRIPTION
**Pull request by jrasell**
The template aims to ensure all PRs include the required information for reviwers. The checklist items help ensure merging happens quickly and in the correct manner.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+merged.
+-->
+
+### All Checklist
+- [ ] **Changelog Entry** If required, please generate and add a changelog entry to this PR using
+  the `make cl` command.
+- [ ] **Testing** Please add tests to cover any new functionality or to demonstrate bug fixes and
+  ensure regressions will be caught.
+- [ ] **Documentation** If the change impacts external components such as the API, please update the
+  Nomad website documentation to reflect this. Please also consider whether the change requires
```
[the remark this answers is not in the document]
Comment (jrasell) [reply .github/pull_request_template.md]: Updated within https://github.com/hashicorp/nomad/pull/24365/commits/f1b026ad8f3726807268437fdb4d21a426835ea8

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+### Links
+<!--
+Please include links to GitHub issues, documentation, or similar which is relevant to this PR. If
+this is a bug fix, please ensure related issues are linked so they will close when this PR is
+merged.
+-->
+
+### All Checklist
+- [ ] **Changelog Entry** If required, please generate and add a changelog entry to this PR using
+  the `make cl` command.
```
[the remark this answers is not in the document]
Comment (jrasell) [reply .github/pull_request_template.md]: Updated within https://github.com/hashicorp/nomad/commit/f1b026ad8f3726807268437fdb4d21a426835ea8

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+-->
+
+### Links
+<!--
+Please include links to GitHub issues, documentation, or similar which is relevant to this PR. If
+this is a bug fix, please ensure related issues are linked so they will close when this PR is
+merged.
+-->
+
+### All Checklist
```
Comment (tgross) [inline .github/pull_request_template.md:18]:
```suggestion
### Contributor Checklist
```

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+- [ ] **Changelog Entry** If this PR changes user-facing behavior, please generate and add a
+  changelog entry using the `make cl` command.
+- [ ] **Testing** Please add tests to cover any new functionality or to demonstrate bug fixes and
+  ensure regressions will be caught.
+- [ ] **Documentation** If the change impacts user-facing functionality such as the CLI, API, UI,
+  and job configuration, please update the  Nomad website documentation to reflect this. Refer to
+  the [website README](../website/README.md) for docs guidelines. Please also consider whether the
+  change requires notes within the [upgrade guide](../website/content/docs/upgrade/upgrade-specific.mdx).
+
+### HashiCorp Employee Checklist
```
Comment (tgross) [inline .github/pull_request_template.md:28]:
```suggestion
### Reviewer Checklist
```

**Pull request merged**